src/Service.Tests/SqlTests/GraphQLPaginationTests/GraphQLPaginationTestBase.cs

```diff
@@ -64,6 +64,92 @@ public async Task RequestFullConnection()
             SqlTestHelper.PerformTestEqualJsonStrings(expected, actual.ToString());
         }
 
+        /// <summary>
+        /// Request a full connection object {items, after, hasNextPage}
+        /// without providing any parameters
+        /// </summary>
+        [TestMethod]
+        public async Task RequestMaxUsingNegativeOne()
+        {
+            string graphQLQueryName = "books";
+            string graphQLQuery = @"{
+                books (first: -1) {
+                    items {
+                        id
+                        title
+                    }
+                    endCursor
+                    hasNextPage
+                }
+            }";
+
+            JsonElement actual = await ExecuteGraphQLRequestAsync(graphQLQuery, graphQLQueryName, isAuthenticated: false);
+            string expected = @"{
+              ""items"": [
+                {
+                  ""id"": 1,
+                  ""title"": ""Awesome book""
+                },
+                {
+                  ""id"": 2,
```

Comment:
updated with comment. On the second point, arent all the other tests in this file susceptible to the same issue on if you change underlying db contents they will get affected as well? If we do a number check also with max page size  (something like number of records returned are <= maxpagesize), we would still need to check that the number of records returned == 12 so that we ensure that it did not end up throwing less rows and that would be susceptible if you add more books to the table

Reply:
Yes but adding is one type of change that we are talking about to a table right? Similarly if I were to remove books which could be another type of change, that would break this case then.

Reply:
we've typically only ever added records to tables that are used in our tests. That should be expected to happen in the future. We don't remove records from tests because of the implications you mentioned, breaking tests which expect certain records to be present.

Reply:
Makes sense will make this fix.

Reply:
Discussed on call. resolving due to large refactoring required to do this specific implementation.